### DISCUSSION TITLE
Zed terminal tries to open a long-removed shell

### DISCUSSION THREAD

**Discussion by tkgalk** [zed-industries/zed#11751] [Feature Requests]
`zsh:1: no such file or directory: /opt/homebrew/bin/fish` Coming up every time I open the Zed terminal. This is the only application that does this.

1. WezTerm opens up my default shell: zsh.
2. VS Code and Jetbrains' open up zsh.
3. Alacritty also opens up fine.
3. Terminal.app opens up zsh.
4. My `.zshrc` has nothing to do with fish.

My full `.zshrc` below.
```
# Preferred editor for local and remote sessions.
if [[ -n $SSH_CONNECTION ]]; then
  export EDITOR='nvim'
else
  export EDITOR='nvim'
fi

# setup zoxide
eval "$(zoxide init --cmd cd zsh)"

# Set up fzf key bindings and fuzzy completion
eval "$(fzf --zsh)"

# Bind nvim to vim.
alias vim=nvim

# Start the starship prompt.
eval "$(starship init zsh)"

# Start fnm.
eval "$(fnm env --use-on-cd)"
```

`.zprofile` file doesn't exist, full `.config/zed/settings.json` below.

```
{
  "theme": "Andromeda",
  "terminal": {
    "dock": "bottom",
    "shell": "system"
  },
  "project_panel": {
    "dock": "right"
  },
  "buffer_font_size": 15,
  "buffer_font_family": "Berkeley Mono",
  "telemetry": {
    "enabled": false
  },
  "ui_font_family": "Berkeley Mono",
  "vim_mode": true
}
```
That shell is not even in `/etc/shells`. No clue how Zed's shell setup is done. Trying to `ripgrep` through the system looking for anything to do with that path turns up nothing (`rg -g '!*.log' -g '!*Library*' -i '/opt/homebrew/bin'`).

Any clues?

**Comment by tkgalk** [comment]
https://github.com/zed-industries/zed/issues/8794#issuecomment-2107500830 could be related or not to this.

**Comment by tkgalk** [comment]
```
2024-05-13T14:17:54+01:00 [ERROR] crates/zed/src/main.rs:214: failed to spawn login shell to source login environment variables

Caused by:
    No such file or directory (os error 2)
2024-05-13T14:17:54+01:00 [INFO] extensions updated. loading 3, reloading 0, unloading 0
2024-05-13T14:17:54+01:00 [ERROR] crates/zed/src/zed.rs:661: EOF while parsing a value at line 1 column 0
```

**Comment by mrnugget** [comment]
Have you tried restarting? I also just played around with shells and there is some state stuck somewhere after changing the default shell that only fixed itself after restarting.

Also, can you run that program that I posted in that comment?

And what do you have in user settings? https://github.com/zed-industries/zed/issues/8794#issuecomment-1976134047

**Reply by tkgalk** [reply]
I've gotten rid of fish months ago, I'm sure I restarted my laptop since then, but I'll give it a go.
$SHELL is `/bin/zsh`, settings in Users & Accounts is `/bin/zsh`.

Running that program in WezTerm and Alacritty returns expected results with `zsh`.

**Reply by tkgalk** [reply]
OMG, restarting DID resolve the Zed issue. I'm confused where that state was kept as ripgrepping found nothing. It works now, though.

**Reply by mrnugget** [reply]
OMG indeed: it's a very special feeling when restarting fixes a problem. I think the proper term is `:lolsob:`
